sunpy/net/dataretriever/tests/test_eve.py

```diff
@@ -1,36 +1,138 @@
 import pytest
+from unittest import mock
+from datetime import datetime, timedelta
 
 from sunpy.time import parse_time
 from sunpy.time.timerange import TimeRange
-from sunpy.net.vso import VSOClient
+from sunpy.net.vso.vso import VSOClient
 from sunpy.net.vso.attrs import Time, Instrument, Source, Level
 from sunpy.net.dataretriever.client import QueryResponse
 import sunpy.net.dataretriever.sources.eve as eve
 from sunpy.net.fido_factory import UnifiedResponse
 from sunpy.net import Fido
+from sunpy.net import attr
 from sunpy.net import attrs as a
 
 
 LCClient = eve.EVEClient()
+BASEURL = eve.BASEURL
+
+
+def create_url(start, end):
+    """
+    This function creates a url based on the EVEClient data,
+    instead of making an online request.
+    """
+
+    start = datetime.strptime(start, '%Y/%m/%d')
+    end = datetime.strptime(end, '%Y/%m/%d')
+
+    lst = list()
+    for n in range(int((end - start).days)+1):
+        lst.append((start+timedelta(n)).strftime(BASEURL))
+    return lst
+
+
+def mock_query_object(start_date, end_date, level=0):
+    """
+    Creation of a QueryResponse object, and prefill some
+    downloaded data from eve.EVEClient().fetch(Time('20 ..)
+    """
+    # Create a mock QueryResponse object
+    map_ = {
+        'TimeRange': TimeRange(parse_time(start_date), parse_time(end_date)),
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'SDO',
+        'instrument': 'eve',
+        'physobs': 'irradiance',
+        'provider': 'LASP'
+    }
+    st_datetime = datetime.strptime(start_date, '%Y/%m/%d')
```

Comment:
IMO, you shouldn't use `datetime` in the codebase XD (Because we have switched to `astropy.time` for our time related stuff and it can do everything `datetime` can).

But maybe @Cadair and @nabobalis might have a different opinion.

Reply:
I tried parsing the date in the `%Y/%m/%d` format in `astropy.time` if I remember it correctly. But it wasn't working in the way as expected. So I switched to `datetime`.

Reply:
If it fails with `astropy.time.Time` it would be good to know, it might be a bug upstream. I would have thought that that would have worked with `astropy.time.Time`. But I have no real issue with using `datetime` in the tests.

Reply:
Let me try that, I would update you with the working.

Reply:
I got this error while parsing - 
```
>>> from astropy import time
>>> a = time.Time('2012/10/4')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/yash/sunpydev/vnv37/lib/python3.7/site-packages/astropy/time/core.py", line 399, in __init__
    precision, in_subfmt, out_subfmt)
  File "/home/yash/sunpydev/vnv37/lib/python3.7/site-packages/astropy/time/core.py", line 454, in _init_from_vals
    precision, in_subfmt, out_subfmt)
  File "/home/yash/sunpydev/vnv37/lib/python3.7/site-packages/astropy/time/core.py", line 505, in _get_time_fmt
    raise ValueError('Input values did not match {0}'.format(err_msg))
ValueError: Input values did not match any of the formats where the format keyword is optional ['datetime', 'iso', 'isot', 'yday', 'datetime64', 'fits', 'byear_str', 'jyear_str']

```